daemon/k8s_watcher.go

```diff
@@ -94,24 +96,63 @@ func k8sErrorHandler(e error) {
 	log.Error(e)
 }
 
-func (d *Daemon) createThirdPartyResources() error {
+func createCustomResourceDefinitions(clientset apiextensionsclient.Interface) error {
```

Comment:
i was also working on some related changes for this. I think we should support both at least until tpr was deprecated fully. Probably something like this should suffice

```
ver, err := k8s.ClusterVersion()
useCRD := len(ver) >= 2 && ver[1] >= 7
// check and use crd, otherwise tpr
```

also we probably want to have a migrator that migrates things from tpr to crd ? not sure if its worth it though or users should just recreate things.